### ISSUE TITLE
Display math $$ delimiter showing up in rendered HTML output

### ISSUE THREAD

**Issue by asmaier** [quarto-dev/quarto-cli#1181]
### Bug description

Here is a (pandoc) markdown test file `test.qmd` 
```
# Test

$$\begin{equation}
1+2
\end{equation}$$
Some text
$$\begin{equation}
3+4
\end{equation}$$
more text
$$\begin{equation}
5+6
\end{equation}$$
even more text
$$\begin{equation}
7+8
\end{equation}$$
the end.
```
When rendering as html with pandoc 

`pandoc -t html5 --mathjax -s -o test-pandoc.html test.qmd`

it renders correctly. 
![Bildschirmfoto 2022-06-19 um 23 04 07](https://user-images.githubusercontent.com/3169297/174500284-cf7061f4-3fce-4284-8699-65e68d3946ee.png)
However when rendering the file to html with quarto 

` quarto render test.qmd`

`$$` are showing up in the text of the html file
![Bildschirmfoto 2022-06-19 um 23 04 23](https://user-images.githubusercontent.com/3169297/174500296-552f1ebd-7f6f-4857-b058-181f31019c69.png)

This happens with quarto 0.9.592 and pandoc 2.18 on Mac OS X 10.15.7 . 

### Checklist

- [X] [formatted your issue](https://yihui.org/issue/#please-format-your-issue-correctly) so it is easier for us to read?
- [X] included a minimal, self-contained, and reproducible example?
- [X] documented the quarto version you're running, by providing the output produced by `quarto check` in a terminal in your issue?
- [X] documented the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] documented which operating system you're running? If on Linux, please provide the specific distribution as well.
- [X] upgraded to the latest version, including your versions of R, the RStudio IDE, and relevant R packages?

**Comment by cscheid** [comment]
Hi - in quarto, you need the `$$` delimiters to be in a line by themselves, because they're used as block delimiters in other parts of the pipeline. The following works:

    # Test
    
    $$
    \begin{equation}
    1+2
    \end{equation}
    $$
    
    Some text
    
    $$
    \begin{equation}
    3+4
    \end{equation}
    $$
    
    more text
    
    $$
    \begin{equation}
    5+6
    \end{equation}
    $$
    
    even more text
    
    $$
    \begin{equation}
    7+8
    \end{equation}
    $$
    
    the end.

**Comment by jjallaire** [comment]
@cscheid I think it would be best if don't introduce special markdown syntax constraints that apply only when using Quarto. So we should ultimately liberalize our parsing rules to handle `$$\begin{equation}` (however I'm fine if we defer that to post v1.0 if we consider the fix to be risky). Let me know what you think.

**Comment by cscheid** [comment]
I'm ok with fixing that, but do note that even in pandoc this doesn't totally work across formats:

```
$ pandoc -t pdf -s -o test-pandoc.pdf test.md
Error producing PDF.
! LaTeX Error: Bad math environment delimiter.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...

l.58 \[\begin{equation}
```

**Comment by jjallaire** [comment]
If that's the case then we definitely shouldn't support it (i.e. if it only works in HTML then we shouldn't encourage it)

**Comment by asmaier** [comment]
> Hi - in quarto, you need the $$ delimiters to be in a line by themselves, because they're used as block delimiters in other parts of the pipeline.

Unfortunately this is not documented here: https://quarto.org/docs/authoring/markdown-basics.html#equations

But apart from that, originally my markdown file was created by converting with pandoc from latex to markdown. So the default markdown writer of pandoc is using `$$` delimiters as shown im the example `test.qmd` above. So if you insist on Quarto needing `$$` on a single line it means people cannot use the default markdown writer of pandoc to produce files which are compatible with Quarto. Of course nowadays you can easily write a lua filter or custom writer in pandoc to solve the issue, but it is still an annoyance for people who might want to convert their files to pandoc markdown for use with Quarto.

**Comment by cscheid** [comment]
> But apart from that, originally my markdown file was created by converting with pandoc from latex to markdown.

Huh, this is interesting. So you have a file which pdflatex builds correctly, but if you convert from latex to markdown with pandoc, then pandoc fails to convert from markdown to pdf? I don't know what are the guarantees that pandoc provides with respect to formats and roundtrips, but I wonder if this is an issue with pandoc.

**Comment by asmaier** [comment]
> So you have a file which pdflatex builds correctly, but if you convert from latex to markdown with pandoc, then pandoc fails to convert from markdown to pdf?

That seemed odd to me and I investigated the issue a bit more. It turns out my example given in the bug description above is not an example that would be created by pandoc when converting from latex. I created what I thought is a minimal example, but in fact it is not quite correct.

When converting from latex 
```
\begin{equation}
1+2
\end{equation} 
Some text 
\begin{equation}
3+4
\end{equation} 
```
this will be converted by pandoc to 
```
$$1+2$$
Some text
$$3+4$$
```
So my example in the bug description is misleading and cannot be converted by pandoc to pdf, as you have shown.

However when starting from a latex file like
```
\begin{align}
1+2
\end{align} 
Some text 
\begin{align}
3+4
\end{align} 
```
this will be converted by `pandoc --wrap=preserve -t markdown -s -o test-mini.md test-mini.tex` to
```
$$\begin{aligned}
1+2\end{aligned}$$
Some text
$$\begin{aligned}
3+4\end{aligned}$$
```
because apparently pandoc is treating the `align` environment different from the `equation` environment. It is converting an `align` environment into `$$\begin{aligned}...\end{aligned}$$` . The latter file however can be converted by pandoc into html and pdf without problems. But when rendering this file with ` quarto render test-mini.md` the HTML will show `$$` in the text:

![Bildschirmfoto 2022-06-21 um 22 12 28](https://user-images.githubusercontent.com/3169297/174889062-789e1c5c-aa6f-4e6f-9890-a8cfcdf609c7.png)

**Comment by rressler** [comment]
Quarto does Not seem to need the $$ to be on a separate line. See https://community.rstudio.com/t/inconsistent-rendering-of-display-math-between-rstudio-and-quarto-render/140669.
I was able to "fix" the issue by getting rid of every line break between the beginning and ending $$. That makes for some painful reading of the LaTeX but the document at least now works. I don't know if it is Quarto or it happens later in the process. However the fact that copied and pasted code will work, then not work, then work again seems strange (as in the reference image.

**Comment by glimmerphoenix** [comment]
I have confirmed that the `$$` delimiter does not need to be on its own line, as @rressler suggests. However, I have also checked that both syntax options cannot be mixed in the same document. That is, with the latest version, one must either type all display style math placing every `$$` delimiter on a separate line or all display math blocks without line breaks between the starting and ending `$$`. If you write any display math block differently, the parser stops working from the first inconsistent math block syntax onward.

Version info:

- Platform: Linux (Ubuntu 20.04).
- RStudio: 2022.07.1 Build 554
- Quarto: 1.0.38
- quarto (CRAN package): 1.2

**Comment by rressler** [comment]
Thanks for checking!
Richard

From: Felipe Ortega ***@***.***>
Reply-To: quarto-dev/quarto-cli ***@***.***>
Date: Tuesday, August 9, 2022 at 12:02 PM
To: quarto-dev/quarto-cli ***@***.***>
Cc: Richard Ressler ***@***.***>, Mention ***@***.***>
Subject: Re: [quarto-dev/quarto-cli] Display math $$ delimiter showing up in rendered HTML output (Issue #1181)


External Email: Use caution with links and attachments.

I have confirmed that the $$ delimiter does not need to be on its own line, as @rressler<https://urldefense.com/v3/__https:/github.com/rressler__;!!IaT_gp1N!ydE_RdFgSl1uKzRYOCvyPyBQAgyaKL4q_c5-DW7H3JwXSnRwTlke5W-7MFK4nUYSPMdDjdXlYIpxhcPt_JyOP84Wt3c$> suggests. However, I have also checked that both syntax options cannot be mixed in the same document. That is, with the latest version, one must either type all display style math placing every $$ delimiter on a separate line or all display math blocks without line breaks between the starting and ending $$. If you write any display math block differently, the parser stops working from the first inconsistent math block syntax onward.

Version info:

  *   Platform: Linux (Ubuntu 20.04).
  *   RStudio: 2022.07.1 Build 554
  *   Quarto: 1.0.38
  *   quarto (CRAN package): 1.2

**Comment by mghassany** [comment]
I'm confused. I tried everything to remove the $$ from the html output in this example and I didn't succeed. What is confusing is that using the same syntax as you can see in the second equation, they do not appear.

```
***Première approche***: Soit $Y=X^2$. La distribution de $Y$ est donnée par

$$
\begin{align}
P(Y=1) &= P(X=-1) + P(X=1) = 0.5 \\
P(Y=0) &= P(X=0) = 0.5
\end{align}$$

Donc $E(X^2)=E(Y) = 1(0.5) + 0(0.5) = 0.5$

***Deuxième approche***: En utilisant le théorème

$$
\begin{align}
E(X^2)  &= (-1)^2(0.2) + 0^2(0.5) + 1^2 (0.3) \\
        &= 1(0.2+0.3)+0(0.5)=0.5
\end{align}$$

Remarquer que $0.5=E(X^2) \neq (E(X))^2 = 0.01$
```

Output: 

![image](https://user-images.githubusercontent.com/11962615/187022168-1cdfdf75-bbbc-4ae0-96f9-af47398b01bc.png)

**Comment by jjallaire** [comment]
This is an issue that has been fixed in our new v1.1 release: https://quarto.org/docs/download/

Try it with v1.1 and it should work as expected.
